cmd/heimdalld/main.go

```diff
@@ -138,6 +184,254 @@ func exportAppStateAndTMValidators(logger log.Logger, db dbm.DB, storeTracer io.
+the flag --bridge when starting Tendermint in process.
+Pruning options can be provided via the '--pruning' flag. The options are as follows:
+
+syncable: only those states not needed for state syncing will be deleted (keeps last 100 + every 10000th)
+nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+everything: all saved states will be deleted, storing only the current state
+
+Node halting configurations exist in the form of two flags: '--halt-height' and '--halt-time'. During
+the ABCI Commit phase, the node will check if the current block height is greater than or equal to
+the halt-height or if the current block time is greater than or equal to the halt-time. If so, the
+node will attempt to gracefully shutdown and the block will not be committed. In addition, the node
+will not be able to commit subsequent blocks.
+
+For profiling and benchmarking purposes, CPU profiling can be enabled via the '--cpu-profile' flag
+which accepts a path for the resulting pprof file.
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !viper.GetBool(flagWithTendermint) {
+				ctx.Logger.Info("starting ABCI without Tendermint")
+				return startStandAlone(ctx, appCreator)
+			}
+
+			ctx.Logger.Info("starting ABCI with Tendermint")
+
+			startRestServer, _ := cmd.Flags().GetBool(helper.RestServerFlag)
+			startBridge, _ := cmd.Flags().GetBool(helper.BridgeFlag)
+
+			_, err := startInProcess(ctx, appCreator, cdc, startRestServer, startBridge)
+			return err
+		},
+		PreRun: func(cmd *cobra.Command, args []string) {
+			// bridge binding
+			if err := viper.BindPFlag("all", cmd.Flags().Lookup("all")); err != nil {
+				logger.Error("GetStartCmd | BindPFlag | all", "Error", err)
+			}
+
+			if err := viper.BindPFlag("only", cmd.Flags().Lookup("only")); err != nil {
+				logger.Error("GetStartCmd | BindPFlag | only", "Error", err)
+			}
+		},
+	}
+
+	cmd.Flags().Bool(
+		helper.RestServerFlag,
+		false,
+		"Start rest server",
+	)
+
+	cmd.Flags().Bool(
+		helper.BridgeFlag,
+		false,
+		"Start bridge service",
+	)
+
+	cmd.PersistentFlags().String(helper.LogLevel, ctx.Config.LogLevel, "Log level")
+	if err := viper.BindPFlag(helper.LogLevel, cmd.PersistentFlags().Lookup(helper.LogLevel)); err != nil {
+		logger.Error("main | BindPFlag | helper.LogLevel", "Error", err)
+	}
+	// bridge flags
+	cmd.Flags().Bool("all", false, "start all bridge services")
+	cmd.Flags().StringSlice("only", []string{}, "comma separated bridge services to start")
+
+	// rest server flags
+	cmd.Flags().String(client.FlagListenAddr, "tcp://0.0.0.0:1317", "The address for the server to listen on")
+	cmd.Flags().Bool(client.FlagTrustNode, true, "Trust connected full node (don't verify proofs for responses)")
+	cmd.Flags().String(client.FlagChainID, "", "The chain ID to connect to")
+	cmd.Flags().String(client.FlagNode, helper.DefaultTendermintNode, "Address of the node to connect to")
```

Comment:
Is this needed? It should already know the target address for the Tendermint node in the code, I think it is one of the flags to bind.

Reply:
This flag was defined for rest-server in cosmos sdk, so it was copied here. Agree, this is for the flag to bind and if not provided it will use default one.